Automattic/woocommerce-payments
# Prevent WooPay order webhook from being registered if account is under review

Context: p1712667655203709-slack-C022WMN88KG

### Changes proposed in this Pull Request

After 4868-gh-Automattic/woocommerce-payments-server, the WooPayments client should expect an `under_review` status after 7.4.0 when the account is suspended, and not a `rejected.*` status.

This PR prevents the WooPay webhook from registering, fixing the regression of a fix previously implemented in https://github.com/Automattic/woocommerce-payments/pull/8118.

### Testing instructions

**Test the regression**

1. Ensure you're in `develop`
2. Ensure WooPay is enabled and working.
3. Change the `account_suspended` column to `1` in the WooPayments server `wcpay_accounts` table, under your local merchant's account ID record.
4. From the merchant store dev tools plugin, clear the account cache.
5. Ensure the `status` prop is `under_review`.
6. From the merchant store, navigate to **wp-admin > WooCommerce > Settings > Advanced > Webhooks**.
7. Ensure you have one active `WCPay woopay order status sync` webhook.
8. Delete the active webhook and refresh the page.
9. Ensure you see an error "Could not process the account." at the `/accounts/platform_checkout` endpoint from WooPayments server's `logstash.log` file.
10. Refresh the Webhooks page and ensure you see a repeated log in WooPayments server's `logstash.log`.

**Test the fix**

1. Checkout to this branch.
2. Refresh the WooCommerce Webhooks page.
3. Ensure there are no new error logs in the WCPay server's `logstash.log`.

-------------------


**Post merge**
